add raster type conversion (#29)
geo-engine/docs: src/operators/rastertypeconversion.md

> # RasterTypeConversion

Reviewer: I guess you also need to add that here: https://github.com/geo-engine/docs/blob/main/src/SUMMARY.md.

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

> The `RasterTypeConversion` operator allows change the data type of raster data.

Reviewer: ```suggestion
The `RasterTypeConversion` operator allows changing the data type of raster data.
```

I recommend installing the Grammarly plugin for VSCode.

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

Author: the plugin does not work :(

It transforms all pixels into the new data type.

> 1. The operator allows loss of precision e.g. converting a `f32` value of "3.1" to a `u8` will return a value of "3".

Reviewer: ```suggestion
1. Applying the operator could lead to a loss of precision, e.g., converting a `F32` value of `3.1` to a `U8` will return a value of `3`.
```

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

> 2. If the old value is not valid in the new type it will clip at the value range of the new type.

Reviewer: Maybe add an example here as well for non computer scientists :smile:

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

## Parameters

> | Parameter | Type                     | Description                  | Example Value         |
> | --------- | ------------------------ | ---------------------------- | --------------------- |
> | `outputDataType` | [`RasterDataType`]| the output type              | "U8"                  |

Reviewer: ```suggestion
| Parameter        | Type               | Description     | Example Value |
| ---------------- | ------------------ | ----------------| ------------- |
| `outputDataType` | [`RasterDataType`] | the output type | `U8`          |
```

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

> The `RasterTypeConversion` operator expects either one _raster_ input.

Reviewer: ```suggestion
The `RasterTypeConversion` operator expects exactly one _raster_ input.
```

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

> | Parameter | Type                         |
> | --------- | ---------------------------- |
> | `source`  | `SingleRasterSource` |

Reviewer: ```suggestion
| Parameter | Type                 |
| --------- | -------------------- |
| `source`  | `SingleRasterSource` |
```

Author: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)
